core/domain/topic_services.py

```diff
@@ -1688,39 +1688,44 @@ def get_topic_id_to_topic_name_dict(topic_ids: List[str]) -> Dict[str, str]:
     return topic_id_to_topic_name
 
 
-def update_chapters_counts_in_topic_summaries(
+def get_chapters_counts_in_topic_summaries(
     topic_summary_dicts: List[topic_domain.FrontendTopicSummaryDict]
-) -> List[topic_domain.FrontendTopicSummaryDict]:
-    """Returns topic summary dicts after updating chapter counts in them.
+) -> Dict[str, topic_domain.TopicChapterCounts]:
+    """Returns topic chapter counts for each topic summary dict.
 
     Args:
         topic_summary_dicts: List[FrontendTopicSummaryDict]. A list of
-            topic summary dicts to be updated with chapter counts.
+            topic summary dicts.
 
     Returns:
-        List[FrontendTopicSummaryDict]. List of updated topic summary dicts.
+        Dict[str, TopicChapterCounts]. Dict of topic id and topic chapter
+        counts domain object.
     """
     topic_ids = [summary['id'] for summary in topic_summary_dicts]
-    topics = topic_fetchers.get_topics_by_ids(topic_ids)
+    topics = list(filter(None, topic_fetchers.get_topics_by_ids(topic_ids)))
```

Comment:
Use a list comprehension rather than a filter (I think we had a lint rule at one point disallowing raw map() and filter(), and I don't think we use filter() anywhere in the codebase).

Ditto elsewhere.

Reply:
Done